src/tree/param.h

```diff
@@ -76,6 +76,8 @@ struct TrainParam : public dmlc::Parameter<TrainParam> {
   int gpu_id;
   // number of GPUs to use
   int n_gpus;
+  // number of rows in a single GPU batch
```

Comment:
I would like to see this be automatically calculated and the parameter removed. It is hard to see a user understanding and manipulating this parameter effectively.

Reply:
Done.